lib/benchpark/runtime.py

```
@@ -82,21 +83,52 @@ def __init__(self, dest):
         self.spack_location = self.dest / "spack"
 
     def bootstrap(self):
+        # Bootstrap Spack first because we will use Spack resources while bootstrapping ramble
+        if not self.spack_location.exists():
+            self._install_spack()
+
+        spack_lib_path = self.spack_location / "lib" / "spack"
+        externals = str(spack_lib_path / "external")
+        if externals not in sys.path:
+            sys.path.insert(1, externals)
+        internals = str(spack_lib_path)
+        if internals not in sys.path:
+            sys.path.insert(1, internals)
+
         if not self.ramble_location.exists():
             self._install_ramble()
+
         ramble_lib_path = self.ramble_location / "lib" / "ramble"
+
+        # If ramble is present but not yet name mangled, do that
+        ramble_spack_path = ramble_lib_path / "ramble_spack"
+        if not ramble_spack_path.exists():
+            self._mangle_ramble()
+
         externals = str(ramble_lib_path / "external")
         if externals not in sys.path:
             sys.path.insert(1, externals)
         internals = str(ramble_lib_path)
         if internals not in sys.path:
             sys.path.insert(1, internals)
 
-        # Spack does not go in sys.path, but we will manually access modules from it
-        # The reason for this oddity is that spack modules will compete with the internal
-        # spack modules from ramble
-        if not self.spack_location.exists():
-            self._install_spack()
+    def _mangle_ramble(self):
+        """Name mangle ramble spack.* symbols to allow separate spack imports."""
+        import llnl.util.filesystem as fs
+
+        files = [
+            f
+            for f in fs.find(self.ramble_location, "*")
+            if os.stat(f).st_mode & stat.S_IWUSR and not os.path.isdir(f)
```

If we don't own these files, we should probably fail, vs. silently doing nothing.

---

I'll add a comment -- the main purpose here isn't to catch files we don't own, it's to catch files that aren't writable (like some things within the `.git` repo).

---

I assume nothing under `.git` should be modified: I recommend skipping it entirely (and explicitly).